libraries/ml/src/distance-matrix/distance-matrix.ts

```diff
@@ -58,8 +58,11 @@ export class DistanceMatrix {
     const size: number = list.length;
     const res = new DistanceMatrix(undefined, size);
     for (let i = 0; i < size; i++) {
-      for (let j = i + 1; j < size; j++)
-        res.set(i, j, method(list[i], list[j]));
+      for (let j = i + 1; j < size; j++) {
+        // if any of the values is null, set distance to 1
+        list[i] !== null && list[j] !== null ?
+          res.set(i, j, method(list[i], list[j])) : res.set(i, j, 1);
```

Comment:
you can do it even better
make the condition inside your set() method - `res.set(i, j, ..... ? method(list[I], list[j]) : 1)`